### DISCUSSION TITLE
Logs: Do NOT output the location of an error in the c++ codebase if it happened strictly in a GDScript file

### DISCUSSION THREAD

**Discussion by jeancallisti** [godotengine/godot-proposals#5789] [Editor]
The whole purpose of GDScript is to write stable code that involves c++ _as little as possible_ .
If you (the game developer) do not dive into c++ plugins, then godot errors almost never happen in the c++ layer of the game, but instead they almost always happen in a GDScript.

**Let's take the simplest possible example :** You write a GDScript that only contains the instruction `assert(false) `  -- which will generate an error every time. **This error does not happen in the c++, only in the GDScript.**

Yet, Godot's logs **always show some error location in the c++ codebase.**
That's completely useless and that clutters the logs **(twice the amount of text!).** Plus, it forces me to read each error **twice** every time, because at first I confuse the c++ location for the relevant info, and I need to read again to see where the issue happened in the GDScript.

Suggestion: Do NOT show in which cpp file the error allegedly happened when the error strictly happened in GDScript.
If that's not possible, create a new option to never write the cpp locations when an error happens.

![image](https://user-images.githubusercontent.com/1881493/201983773-51249412-9840-4cfd-9d74-b299dfd2fa37.png)

**Comment by Calinou** [comment]
This could be done by having a special case to hide the `at:` line for specific methods within C++ files (such as `call` in `modules/gdscript/gdscript_functions.cpp`, but it's bound to have some false positives if not done carefully.

I think this is already done in the Debugger > Errors tab of the editor.

**Reply by jeancallisti** [reply]
_"I think this is already done in the Debugger > Errors tab of the editor."_ 

...Are you saying this feature is already implemented?

**Reply by Calinou** [reply]
> ...Are you saying this feature is already implemented?

Yes, but only in the editor GUI, not in the command line output.
